src/pepr/operator/controllers/network/authorizationPolicies.ts

```diff
@@ -424,21 +424,17 @@ export function findMatchingSsoClient(
     return undefined;
   }
 
-  return pkg.spec.sso?.find(sso => {
-    const waypointSelector = sso.enableAuthserviceSelector;
-
-    // Skip if no selector defined
-    if (waypointSelector === undefined || waypointSelector === null) {
-      return false;
-    }
+  const authserviceClients = getAuthserviceClients(pkg);
+  return authserviceClients.find(client => {
+    const waypointSelector = client.enableAuthserviceSelector;
 
     // Empty object means namespace-wide protection
-    if (Object.keys(waypointSelector).length === 0) {
+    if (Object.keys(waypointSelector!).length === 0) {
       return true;
     }
 
     // For non-empty selectors, check if any label matches
-    return Object.entries(waypointSelector).some(([key, value]) => selector[key] === value);
+    return Object.entries(waypointSelector!).some(([key, value]) => selector[key] === value);
```

Comment:
I think maybe this should be `.every` rather than `.some`? The equivalent label check for waypoints uses `.every` - https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/operator/controllers/istio/waypoint-utils.ts#L74